src/editors/containers/VideoEditor/components/VideoSettingsModal/components/TranscriptWidget/hooks.js

```diff
@@ -11,19 +13,57 @@ export const transcriptLanguages = (transcripts) => {
   return 'None';
 };
 
-export const fileInput = () => {
+export const hasTranscripts = (transcripts) => {
+  if (transcripts && Object.keys(transcripts).length > 0) {
+    return true;
+  }
+  return false;
+};
+
+export const onSelectLanguage = ({ fileName }) => (e) => {
+  const currentTranscripts = useSelector(selectors.video.transcripts);
+  const newTranscripts = { [e.target.value]: { fileName }, ...currentTranscripts };
+  const dispatch = useDispatch();
+  dispatch(actions.video.updateField({ transcripts: newTranscripts }));
+};
+
+export const replaceFileCallback = ({ language, dispatch }) => (e) => {
+  const selectedFile = e.target.files[0];
+  dispatch(thunkActions.video.replaceTranscript({ newFile: selectedFile, newFilename: selectedFile.name, language }));
+};
+
+export const fileInput = ({ onAddFile }) => {
+  const dispatch = useDispatch();
   const ref = React.useRef();
   const click = () => ref.current.click();
   const addFile = (e) => {
-    const selectedFile = e.target.files[0];
-    console.log(selectedFile);
+    if (typeof onAddFile === 'function') {
+      onAddFile(e);
+    } else {
+      const selectedFile = e.target.files[0];
+      dispatch(thunkActions.video.uploadTranscript({
+        filename: selectedFile.name,
+        language: 'esp',
```

Comment:
we might want to add something more sophisticated here as to prevent collision-- as to prevent lang overlap we can grab from selector openLanguages and grab the first one.

Reply:
I think that would be part of the laundry list of little things we can do to later improve the feature.